### ISSUE TITLE
Moving from grunt to gulp

### ISSUE THREAD

**Issue by softwarespot** [jpederson/Squirrel.js#29]
I was recently persuaded to move from grunt to gulp and was wondering is this library tied to grunt? I could easily port over to gulp if you feel it's worth the change.

I personally found that it's a lot quicker and easier to customise.

**Comment by jpederson** [comment]
Hey, sorry I didn't respond to this - I'm not opposed to a switch, I've never messed with Gulp, but I'm certainly open to it and have heard good things.

**Comment by softwarespot** [comment]
Hey, no worries. =) I closed it because I just thought grunt is fine for this project., though I can come up with something at least in the next couple of days.

**Comment by softwarespot** [comment]
Have a look at this fork of [Creep.js](https://github.com/softwarespot/Creep.js) I have been messing around with. You will see that I migrated from Grunt to gulp and it was very easy.

Edit: I have created a new branch called [gulp](https://github.com/jpederson/Squirrel.js/tree/gulp) for you to play around with.

**Comment by softwarespot** [comment]
If you check the [gulp](https://github.com/jpederson/Squirrel.js/tree/gulp) fork, you will notice I have now added a package to update the package.json version number based on the version number in squirrel.js. This in itself reduces time as well as forgetting to update package.json. About publishing to npm, a gulp package can do this too.

**Comment by softwarespot** [comment]
So a lot has taken place in the following branch, one of which is it's now using ESLint for both linting and format verification using the Airbnb style guide.

**Comment by jpederson** [comment]
I'd rather hand-pick the releases to NPM since they aren't always necessary if we aren't making changes to the actual JS, but that's just my preference. I don't mind the version bumping bit though, so that's cool. As for Airbnb style guide, I've never really been a huge fan of it, but I do know that most people use camel case and not underscores, so I suppose that change is fine, and maybe a major version upgrade/refactor would be a good idea to roll that into the project in the future.

A little note about changes to this and master though, I want to switch to a branch development scheme rather than pushing stuff straight to master. It helps separate the changes we're making so that things are easier to track. It's time consuming to go through every single commit when you're making lots of changes to a single branch that I thought was really only suppose to switch to gulp (style guide updates and refactors of code should be in different branches so that we can roll in only the gulp update and not all the other changes if we want to - make sense?).

**Comment by jpederson** [comment]
Also, feel free to develop your branches inside this repo if you like - I feel like this is now *our* project, so collaborating in the same place is cool, and there are benefits we'll get from having our branches in the same repo as the master.

**Comment by softwarespot** [comment]
Yeah, my apologies for going off the rails a little bit with the master branch. You're right though, a beta/dev branch should have been created, though the commits I pushed to master weren't "gulp" related, but none the less they were many.
